src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -1099,14 +1210,81 @@ boolean accept(FrameDesc source, byte[] buf, int off, int len)
 
                     if (tsTranslation != null)
                     {
-                        long srcTs = pktOut.getTimestamp();
                         long dstTs = tsTranslation.apply(srcTs);
 
                         if (dstTs != srcTs)
                         {
                             pktOut.setTimestamp(dstTs);
                         }
                     }
+
+                    if (dstPictureID > -1)
+                    {
+                        MediaStreamTrackDesc source = weakSource.get();
+                        assert source != null;
+
+                        REDBlock redBlock = source.getMediaStreamTrackReceiver()
+                            .getStream().getPayloadBlock(
+                                pktOut.getBuffer(),
+                                pktOut.getOffset(),
+                                pktOut.getLength());
+
+                        if (!DePacketizer
+                            .VP8PayloadDescriptor.hasExtendedPictureId(
+                            redBlock.getBuffer(), redBlock.getOffset(),
+                            redBlock.getLength()))
+                        {
+                            // XXX we have observed that using a non-extended
```

Comment:
Is this meant as a temporary fix? It worries me because it executes on each video packet, allocates new memory and copies the payload. It alo assumes that the VP8 payload descriptor has a certain format (already contains a short PictureID).